client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/regression/MeanSquaredLogarithmicErrorMetric.java

```diff
@@ -0,0 +1,142 @@
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Calculates the mean squared error between two known numerical fields.
+ *
+ * equation: msle = 1/n * Σ(log(y + offset) - log(y´ + offset))^2
+ * where offset is used to make sure the argument to log function is always positive
+ */
+public class MeanSquaredLogarithmicErrorMetric implements EvaluationMetric {
+
+    public static final String NAME = "mean_squared_logarithmic_error";
+
+    public static final ParseField OFFSET = new ParseField("offset");
+
+    private static final ConstructingObjectParser<MeanSquaredLogarithmicErrorMetric, Void> PARSER =
+        new ConstructingObjectParser<>(NAME, true, args -> new MeanSquaredLogarithmicErrorMetric((Double) args[0]));
+
+    static {
+        PARSER.declareDouble(optionalConstructorArg(), OFFSET);
+    }
+
+    public static MeanSquaredLogarithmicErrorMetric fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    private final Double offset;
+
+    public MeanSquaredLogarithmicErrorMetric(@Nullable Double offset) {
+        this.offset = offset;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        if (offset != null) {
+            builder.field(OFFSET.getPreferredName(), offset);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        MeanSquaredLogarithmicErrorMetric that = (MeanSquaredLogarithmicErrorMetric) o;
+        return Objects.equals(this.offset, that.offset);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(offset);
+    }
+
+    public static class Result implements EvaluationMetric.Result  {
+
+        public static final ParseField ERROR = new ParseField("error");
+        private final double error;
+
+        public static Result fromXContent(XContentParser parser) {
+            return PARSER.apply(parser, null);
+        }
+
+        private static final ConstructingObjectParser<Result, Void> PARSER =
+            new ConstructingObjectParser<>("mean_squared_error_result", true, args -> new Result((double) args[0]));
+
+        static {
+            PARSER.declareDouble(constructorArg(), ERROR);
+        }
+
+        public Result(double error) {
+            this.error = error;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+            builder.startObject();
+            builder.field(ERROR.getPreferredName(), error);
```

Comment:
The field name "error" is confusing could this be "msle"

Reply:
Actually I see MeanSquaredError also has a field "error" and in the context of the result object it makes sense